splunk_connect_for_snmp/inventory/tasks.py

```diff
@@ -222,3 +255,91 @@ def is_smart_profile_valid(profile_name, profile):
         logger.warning(f"Patterns for profile {profile_name} must be a list")
         return False
     return True
+
+
+def filter_condition_on_database(mongo_client, address: str, conditions: list):
+    attributes = mongo_client.attributes
+    query = create_query(conditions, address)
+    result = attributes.find(
+        query, {"address": 1, "group_key_hash": 1, "_id": 0, "indexes": 1}
+    )
+    return list(result)
+
+
+def create_profile(profile_name, varBinds, records):
+    varbind_list = []
+    for record in records:
+        for varbind in varBinds:
+            varbind_list.append(varbind.copy() + record["indexes"])
+    profile = {profile_name: {"frequency": 10, "varBinds": varbind_list}}
+    return profile
+
+
+def create_query(conditions: typing.List[dict], address: str) -> dict:
+
+    conditional_profiles_mapping = {
+        "equals": "$eq",
+        "gt": "$gt",
+        "lt": "$lt",
+        "in": "$in",
+    }
+
+    def _parse_mib_component(field: str) -> str:
+        mib_component = field.split("|")
+        if len(mib_component) < 2:
+            raise BadlyFormattedFieldError(f"Field {field} is badly formatted")
+        return mib_component[0]
+
+    def _convert_to_float(value: str, ignore_error=False) -> typing.Any:
```

Comment:
This is actually nice implementation.I would only check before the conversion to float(value) whether it is numeric:
if(value.isnumeric())
   return float(value)
you can also use try-except-else block if you want to avoid catching the ValueError exception.

Reply:
Yes, but here I have two cases - in one I should ignore that the value can't be float (when there is an in operator that can have any value) and the one where I should inform user that he configured profile wrongly - when he used lt or gt that can be only numeric. Not sure how try-except-else block would help me in that. I wouldn't go for the `isnumeric()` option as it has many limitations - for example it won't work with decimal points. For ex:
```
>>> "1.2".isnumeric()
False
```